vmsdk/rust/cctrusted/src/api_data.rs

```diff
@@ -0,0 +1,65 @@
+/***
+ ************************************
+ * API get_cc_report() related data *
+ ************************************
+ */
+
+// input of API get_cc_report()
+// this struct is used in vTPM and other TEE scenarios
+// e.g.: vTPM may need report based on selective PCRs
+pub struct ExtraArgs {}
+
+pub const TYPE_PLAIN: i8 = -1;
+pub const TYPE_TDX: i8 = 0;
+pub const TYPE_SEV: i8 = 1;
+pub const TYPE_CCA: i8 = 2;
+pub const TYPE_TPM: i8 = 3;
```

Comment:
why we cannot reuse the previous definition?

Code duplication is not a good idea.

Reply:
The initial definition of these types are at cctruste_base crate:(see https://github.com/hairongchen/cc-trusted-api/blob/rust_sdk_framework/common/rust/cctrusted_base/src/cc_type.rs#L12) which should not be refernced directly by API use, instead, as the VM SDK user, it should reference data structures and functions from cctrused crate, so the type is redefined in VM SDK to make sure API user can access necessary data structure for its further usage.

Reply:
Please help me understand why "cctruste_base crate should not be refernced directly by API user".

I am confused because I do see a strong relationship between them. E.g. when we change TPM type to 0, we need to update both places.

Reply:
the initial idea is we want API SDK user can use as less packages as possible in the usage, but after an discussion, I think your suggestion is reasonable, so it is updated to make sure the data structure define in cctrused_base can be shared in both SDK and API SDK user, e.g.: https://github.com/hairongchen/cc-trusted-api/blob/rust_sdk_framework/vmsdk/rust/sample/src/cc-sample.rs#L1